bootloaders/riotboot_dfu/main.c

```diff
@@ -27,6 +27,22 @@
 #include "riotboot/slot.h"
 #include "riotboot/usb_dfu.h"
 
+#include "bootloader_selection.h"
+
+#ifdef BTN_BOOTLOADER_PIN
+#include "periph/gpio.h"
+#endif
+
+static bool bootloader_alternative_mode(void)
+{
+#ifdef BTN_BOOTLOADER_PIN
+    gpio_init(BTN_BOOTLOADER_PIN, BTN_BOOTLOADER_MODE);
+    return !!gpio_read(BTN_BOOTLOADER_PIN) != BTN_BOOTLOADER_INVERTED;
```

Comment:
If the double `!!` is needed, can we add some parenthesis around ? I find this terrible to read.

Reply:
It is needed as gpio_read only promises results `>0` and the comparison would receive promotion.

However, I think I can do even better than adding parentheses and explicitly cast to bool -- and unless you still prefer, would not add extra parens around that.

Reply:
That's enough for me now, this is much clearer. Thanks !